object_intern.go

```diff
@@ -282,14 +217,18 @@ func (oi *ObjectIntern) AddOrGetStringCompressed(obj []byte) (string, error) {
 		// increment reference count by 1
 		(*(*uint32)(unsafe.Pointer(addr + uintptr(len(objComp)))))++
 		tmpString := objString
-		StringHeader := (*reflect.StringHeader)(unsafe.Pointer(&tmpString))
-		StringHeader.Data = addr
+		(*reflect.StringHeader)(unsafe.Pointer(&tmpString)).Data = addr
```

Comment:
what's the point of allocating `tmpString` instead of just using `objString` and set it's `.Data` and then return it?